lib/lib-rds/src/Signer.ts

```diff
@@ -0,0 +1,61 @@
+import { Sha256 } from "@aws-crypto/sha256-js";
+import { defaultProvider } from "@aws-sdk/credential-provider-node";
```

Comment:
This makes the package requiring all the Node.js specific package, making it not compatible with mobile and browser runtime. Also the `Sha256` is default to pure JS implementation. One thing you can do add a runtime-specific replacement for it. You can look into how it's done in clients:
* [Specify browser and react-native entry in package.json](https://github.com/aws/aws-sdk-js-v3/blob/b0e1422630bbdc6842f7b91629d1ff66693e0fde/clients/client-accessanalyzer/package.json#L84-L89)
* [Throwing error in browsers if credentials are missing](https://github.com/aws/aws-sdk-js-v3/blob/b0e1422630bbdc6842f7b91629d1ff66693e0fde/clients/client-accessanalyzer/src/runtimeConfig.browser.ts#L29)
* [Use `@aws-crypto/sha256-browser` in browser and RN runtimes](https://github.com/aws/aws-sdk-js-v3/blob/b0e1422630bbdc6842f7b91629d1ff66693e0fde/clients/client-accessanalyzer/src/runtimeConfig.browser.ts#L37)
* [Use `@aws-sdk/hash-node` in Node.js runtime for SHA256 util](https://github.com/aws/aws-sdk-js-v3/blob/b0e1422630bbdc6842f7b91629d1ff66693e0fde/clients/client-accessanalyzer/src/runtimeConfig.ts#L46)